francis/model_adaptor.py

```diff
@@ -25,6 +25,23 @@ def adapt_predictions(predictions, categories):
     return list(map(lambda x: categories[x], predictions_as_num))
 
 
+def summarize_predictions(predictions, categories, seconds=5):
```

Comment:
I like the sumarising of this, but I think this be kept as two funtions.
1. producing a list of predictions
2. summarising that list of predictions

one is data transformation and one is data display
**each function should have a single responsibility**
you did this very well with the spinner class, and using the with block 😃 

https://en.wikipedia.org/wiki/Single-responsibility_principle